pkg/prometheus/store.go

```diff
@@ -0,0 +1,90 @@
+// Copyright 2023 The prometheus-operator Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package prometheus
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/pkg/errors"
+
+	monv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	"github.com/prometheus-operator/prometheus-operator/pkg/assets"
+)
+
+func AddRemoteWritesToStore(ctx context.Context, store *assets.Store, namespace string, remotes []monv1.RemoteWriteSpec) error {
+	if store == nil && len(remotes) > 0 {
```

Comment:
I wouldn't do defensive programming. If store is nil, we have a bug and it's better to crash.

Reply:
Hey, taking this opportunity to learn a bit more about the operator, what exactly are the stores? Why would it be a bug if they are nil?

Reply:
How about we return an error rather than crash? I feel something odd about terminating a program abruptly, especially when we can handle that state gracefully ?

Reply:
a nil store means that the developer forgot to pass a valid instance which is always a bug.

Reply:
A store is a struct where we keep data referenced by secret or configmap keys such as TLS certificates, bearer tokens, .... If the same secret key is referenced by several resources, the operator will get it once from the Kube API. It also serves as a way to drop resources that reference non-existing data.